pyproject.toml

```diff
@@ -32,6 +32,7 @@ beets = ">=1.4,<=2"
 httpx = ">=0.27.0"
 ordered-set = ">=4.0"
 packaging = ">=24.0"
+requests = "^2.32.3"
```

Comment:
Is there a need for this? We recently migrated to `httpx`.

Reply:
See my comment on the original issue:
https://github.com/snejus/beetcamp/issues/66#issuecomment-2366778064

Reply:
Have replied to your comment there!

Reply:
Again see:
https://github.com/snejus/beetcamp/issues/66#issuecomment-2382744892

Reply:
Thanks for this, you raised a very good point. Instead of adding an explicit dependency here, try adjusting the `beets` dependency as follows:

```toml
beets = { version = ">=1.5,<=2", extras = ["fetchart"] }
```

Reply:
updated the PR :)

Reply:
updated the PR :)

Reply:
I am not sure if the test use the poetry.lock file or just create an env from scratch because locally using the lock file I just fixed everything passes as expected but if i upgrade beets to 2.0.0 I get the same errors locally...

Reply:
Did you `poetry install` after updating `poetry.lock`?

Also try rebasing off the `main` branch now. I expect you will have a conflict in `poetry.lock`: feel free to use this shell function I have to resolve such conflicts

```fish
porb() {
  # args:
  # info: fix _poetry.lock conflicts
  git restore --staged --worktree poetry.lock
  poetry lock -v --no-update
  git add poetry.lock
}
```

Reply:
Yes I did but I think the issue is related to a change in what tags beets is using. The diff of locally run pytests with beets 2.0.0 looks like this:
```log
Full diff:
    {
        'album': 'Common Assault EP',
        'album_id': 'https://foldrecords.bandcamp.com/album/fld001-gareth-wild-common-assault-ep',
        'albumdisambig': None,
        'albumstatus': 'Official',
        'albumtype': 'ep',
        'albumtypes': [
            'ep',
        ],
        'artist': 'Gareth Wild',
        'artist_credit': None,
        'artist_id': 'https://foldrecords.bandcamp.com',
        'artist_sort': None,
  +     'artists': [],
  +     'artists_credit': [],
  +     'artists_ids': [],
  +     'artists_sort': [],
        'asin': None,
  +     'barcode': None,
        'catalognum': 'FLD001',
        'comments': None,
        'country': 'GB',
        'data_source': 'bandcamp',
        'data_url': 'https://foldrecords.bandcamp.com/album/fld001-gareth-wild-common-assault-ep',
        'day': 2,
        'discogs_albumid': None,
        'discogs_artistid': None,
        'discogs_labelid': None,
        'genre': None,
        'label': 'FOLD RECORDS',
        'language': None,
        'media': 'Digital Media',
        'mediums': 1,
        'month': 11,
        'original_day': None,
        'original_month': None,
        'original_year': None,
  +     'release_group_title': None,
        'releasegroup_id': None,
        'releasegroupdisambig': None,
        'script': None,
        'style': None,
        'tracks': None,
        'va': False,
        'year': 2020,
    }
ERROR tests/test_plugin.py::test_album_for_id[album-Vinyl-Vinyl] - KeyError: 'artists'
ERROR tests/test_plugin.py::test_album_for_id[album--Digital Media] - KeyError: 'artists'
ERROR tests/test_plugin.py::test_singleton_candidates[single_track_release] - KeyError: 'artists'
ERROR tests/test_plugin.py::test_candidates[album] - KeyError: 'artists'
ERROR tests/test_plugin.py::test_album_for_id[album-CD-Digital Media] - KeyError: 'artists'
```

Reply:
Okay, this makes sense, I see you ran `poetry update` instead of `poetry lock`. 

Note: the `update` command looks for the most recent dependency version, while `lock` simply synchronises lock file with changes in `pyproject`, keeping `beets` version `1.6.0` in this case.

It's my bad though - `beets` dependency constraint was supposed to be `">=1.4,<2"` instead of `">=1.4,<=2"` - feel free to adjust it and re-run `poetry update` - this issue should be gone.

Reply:
I did run `poetry lock` but did not pass the `--no-update` flag which changed the versions in the lock file that then were applied with `poetry install`. The breaking change however occurred from `beets v1.6.1` onwards. I modified the dependency (in #69 as well) to `">=1.4,<=1.6"` which worked with the existing test framework.